tools/profileBuilder/cmd/list.go

```diff
@@ -54,65 +49,50 @@ $> ../model/testdata/smallProfile.txt > profileBuilder list --name small_profile
 	Args: cobra.NoArgs,
 	Run: func(cmd *cobra.Command, args []string) {
 		logWriter := ioutil.Discard
-		if viper.GetBool("verbose") {
+		if verboseFlag {
 			logWriter = os.Stdout
 		}
 
 		outputLog := log.New(logWriter, "[STATUS] ", 0)
 		errLog := log.New(os.Stderr, "[ERROR] ", 0)
 
-		outputLog.Printf("Output-Location set to: %s", viper.GetString(outputLocationLongName))
+		if !filepath.IsAbs(outputRootDir) {
+			abs, err := filepath.Abs(outputRootDir)
+			if err != nil {
+				errLog.Fatalf("failed to convert to absolute path: %v", err)
+			}
+			outputRootDir = abs
+		}
+		outputLog.Printf("Output-Location set to: %s", outputRootDir)
 
-		var input io.Reader
+		inputFile, err := cmd.Flags().GetString(inputLongName)
+		if err != nil {
+			errLog.Fatalf("failed to get %s: %v", inputLongName, err)
+		}
 
-		if _, ok := listFlags.Get(inputLongName).(int); ok {
-			input = os.Stdin
-		} else if fileHandle, err := os.Open(listFlags.GetString(inputLongName)); err == nil {
-			input = fileHandle
-		} else {
-			errLog.Printf("Fatal! Unable to open file %q", listFlags.GetString(inputLongName))
-			return
+		data, err := ioutil.ReadFile(inputFile)
+		if err != nil {
+			errLog.Fatalf("failed to read list: %v", err)
 		}
 
-		deleteLoc := path.Join(listFlags.GetString(outputLocationLongName), listFlags.GetString(nameLongName))
-		if viper.GetBool("clear-output") {
-			if err := model.DeleteChildDirs(deleteLoc); err != nil {
-				errLog.Print("Fatal! Unable to clear output-folder:", err)
-				return
+		var listDef model.ListDefinition
+		err = json.Unmarshal(data, &listDef)
+		if err != nil {
+			errLog.Fatalf("failed to unmarshal JSON: %v", err)
+		}
+
+		if clearOutputFlag {
+			if err := model.DeleteChildDirs(outputRootDir); err != nil {
+				errLog.Fatalf("Unable to clear output-folder: %v", err)
 			}
```

Comment:
should we fail in this case ?

Reply:
Call to `Fatalf` will log the message then call `os.Exit(-1)`, or are you saying this shouldn't be considered a fatal error?

Reply:
well so yea, was asking if that is what we want, to just fail or warn and continue

---
In reply to: [238000977](https://github.com/Azure/azure-sdk-for-go/pull/3440#discussion_r238000977) [](ancestors = 238000977)

Reply:
hmm I guess if the user put that flag, I guess its fair enough to fail since we are not doing what he requested 

---
In reply to: [238003925](https://github.com/Azure/azure-sdk-for-go/pull/3440#discussion_r238003925) [](ancestors = 238003925,238000977)

Reply:
The way it works right now it would be safe to warn and continue since we only write the one file with the same name (well, that assumes that we open the file for truncate).  However this could be problematic in the future if we ever emit more than one file and/or the file name can change.